packages/lib/src/ContentContainer.wc.svelte

```diff
@@ -0,0 +1,104 @@
+<svelte:options customElement={{ tag: 'sesamy-content-container', shadow: 'open' }} />
+
+<script lang="ts">
+  import type { SesamyAPI, Entitlement } from '@sesamy/sesamy-js';
+  import Base from './Base.svelte';
+  import type { ContentContainerProps } from './types';
+  let {
+    'item-src': itemSrc = '',
+    'publisher-content-id': publisherContentId,
+    'access-url': accessUrl = '',
+    pass = '',
+    'access-level': accessLevel = 'entitlement',
+    'lock-mode': lockMode = 'embed'
+  }: ContentContainerProps = $props();
+
+  async function checkAccess(api: SesamyAPI) {
```

Comment:
This is a bit tricky and probably should be made clearer with an explicit return type. It can return either an entitlement (the first matching) or true. If for instance the lock-mode is set to `logged-in` there's no point in matching the entitlements, so in those cases we will just return true.